bot/exts/fun/fun.py

```diff
@@ -153,10 +154,28 @@ async def caesarcipher_decrypt(self, ctx: Context, offset: int, *, msg: str) ->
         await self._caesar_cipher(ctx, offset, msg, left_shift=True)
 
     @commands.command()
-    async def joke(self, ctx: commands.Context, category: Literal["neutral", "chuck", "all"] = "all") -> None:
-        """Retrieves a joke of the specified `category` from the pyjokes api."""
-        joke = pyjokes.get_joke(category=category)
-        await ctx.send(joke)
+    async def joke(self, ctx: commands.Context, category: Literal["dad", "neutral", "chuck", "all"] = "all") -> None:
+        """
+        Retrieves a joke of the specified `category` from the pyjokes api.
+
+        - dad uses icanhazdadjoke.
+        - others use pyjokes.
+        """
+        if category == "dad":
+            async with aiohttp.ClientSession() as session, session.get(
+                "https://icanhazdadjoke.com",
+                headers={
+                    "Accept":"application/json",
+                    "User-Agent": "Sir-lancebot"
+                }) as res:
```

Comment:
- Reusing the bot's internal http_session is more efficient than creating a new one.
- Including a contact method (e.g. the URL to the project) in the user agent is good practice.
```suggestion
            async with self.bot.http_session.get(
                "https://icanhazdadjoke.com",
                headers={
                    "Accept": "application/json",
                    "User-Agent": "Sir-Lancebot (https://github.com/python-discord/sir-lancebot)",
                }
            ) as res:
```